### ISSUE TITLE
Avoid adding 'Input' in case of using INPUT_OBJECT annotation

### ISSUE THREAD

**Issue by adammach** [ExpediaGroup/graphql-kotlin#1324]
Hi,
is there any possibility to avoid adding 'Input' postfix to name of input objects? If object is already with INPUT_OBJECT annotation I think this is not necessarily. 

thanks

**Comment by mervyn-mccreight** [comment]
Is this referring to schema generation? So you mean there should not be a post-fixed "Input" to class-names that are annotated with `INPUT_OBJECT` in the generated schema?

Or what is the desired behavior in this issue?
I'm asking because I'm thinking about contributing a PR :)

**Comment by dariuszkuc** [comment]
Hello 👋 
@mervyn-mccreight yes. When object is annotated with `INPUT_OBJECT` as the only valid location, it is not possible to reuse it as an output type hence there is no need to explicitly add the suffix there (suffix is added by default to avoid collisions when same Kotlin type is used for both input and output types).

**Comment by mervyn-mccreight** [comment]
Unfortunately I forgot to mention inside the PR description that it closes this issue, so one has to close it by hand I am afraid 😅

**Comment by dariuszkuc** [comment]
No worries, thanks for the contribution!

**Comment by bherrmann2** [comment]
What if an object needs to be changed to be both an input and output object? The same one now cannot be used.

**Comment by dariuszkuc** [comment]
> What if an object needs to be changed to be both an input and output object? The same one now cannot be used.

While you can use same object in the code for both input and output type, GraphQL spec makes a distinction between input and output types and they have to have unique names. That is why if type is used for both input and output types, we append `Input` suffix to make a distinction between them. 

If you define `Foo` as input type in your schema, you won't be able to use the same name for output types - if you were to update locations to support both then your input type in the schema would change to `FooInput` which is a potentially breaking schema change.

**Comment by bherrmann2** [comment]
Yes exactly. Is that something we're ok with? Before this change, you could change an object from input only to both input and output without any potential breaking change

**Comment by samuelAndalon** [comment]
https://github.com/ExpediaGroup/graphql-kotlin/pull/1438
